### PR TITLE
feat: add shop button to collect on the secondary marketplace

### PR DESCRIPTION
**Pull request by ahomentc**
Add "Shop Collects" button to Lenstube (like the one on Lenster).

What does this PR do?
Shop button will allow users to find a video's collects on the secondary marketplace. If a video is sold out of collects, users might want to buy one from secondary. This makes it easy for them to find those collects they want to buy.

Instead of browsing on LensPort for videos to collect they can also browse on Lenstube now.

image

Type of change
 New feature (non-breaking change which adds functionality)
 Enhancement (non-breaking small changes to existing functionality)

### CODE REVIEW

**Review by sasicodes — CHANGES_REQUESTED** [review]

File: src/components/Watch/CollectVideo/CollectModal.tsx [hunk 1]
```diff
@@ -177,37 +179,41 @@ const CollectModal: FC<Props> = ({
...
+                  <PermissionAlert
+                    isAllowed={isAllowed}
+                    setIsAllowed={setIsAllowed}
+                    allowanceModule={
+                      allowanceData
+                        ?.approvedModuleAllowanceAmount[0] as ApprovedAllowanceAmount
+                    }
+                  />
+                )}
+                <Button onClick={() => shopCollects()}>Shop Collects</Button>
```
[the remark this answers is not in the document]
Comment (ahomentc) [reply src/components/Watch/CollectVideo/CollectModal.tsx:214]: Modified to hide the button

File: src/components/Watch/CollectVideo/index.tsx [hunk 2]
```diff
@@ -157,6 +157,26 @@ const CollectVideo: FC<Props> = ({ video, variant = 'primary' }) => {
...
+        return
+      }
+      pubId = video.mirrorOf.id
+    } else {
+      pubId = video.id
+    }
+    const decimalProfileId = parseInt(pubId.split('-')[0], 16)
+    const decimalPubId = parseInt(pubId.split('-')[1], 16)
+    const marketplacePublicationId = decimalProfileId + '_' + decimalPubId
+    const marketplaceUrl = 'http://lensport.io/p/' + marketplacePublicationId
```
[the remark this answers is not in the document]
Comment (ahomentc) [reply src/components/Watch/CollectVideo/index.tsx]: Added the website URL to constants.ts. There isn't a public URL for testnet version of marketplace yet. Will come back and add that when testnet is live.

File: src/components/Watch/CollectVideo/index.tsx [hunk 3]
```diff
@@ -157,6 +157,26 @@ const CollectVideo: FC<Props> = ({ video, variant = 'primary' }) => {
     })
   }
 
+  const shopCollects = () => {
+    if (!video) {
+      return
+    }
+    var pubId = ''
```
[the remark this answers is not in the document]
Comment (ahomentc) [reply src/components/Watch/CollectVideo/index.tsx]: Modified to be cleaner like this

**Review by sasicodes — APPROVED** [review]
Yugi Thumb

**Comment by sasicodes** [comment]
Thanks for the contribution! @ahomentc @gitpoap-bot

**Pull request merged**